server/src/main/java/org/opensearch/search/internal/ContextIndexSearcher.java

```diff
@@ -98,15 +98,31 @@ public class ContextIndexSearcher extends IndexSearcher implements Releasable {
     private QueryProfiler profiler;
     private MutableQueryTimeout cancellable;
 
+    /**
+     * Certain queries can benefit if we reverse the segment read order,
+     * for example time series based queries if searched for desc sort order
+     */
+    private final boolean reverseLeafReaderContexts;
+
     public ContextIndexSearcher(
         IndexReader reader,
         Similarity similarity,
         QueryCache queryCache,
         QueryCachingPolicy queryCachingPolicy,
         boolean wrapWithExitableDirectoryReader,
-        Executor executor
+        Executor executor,
+        boolean reverseLeafReaderContexts
```

Comment:
Lets add a new ctor so `reverseLeafReaderContexts` isn't a required boolean. That'll minimize the test changes above.

```
    public ContextIndexSearcher(
        IndexReader reader,
        Similarity similarity,
        QueryCache queryCache,
        QueryCachingPolicy queryCachingPolicy,
        boolean wrapWithExitableDirectoryReader,
        Executor executor) {
    this(reader, similarity, queryCache, queryCachingPolicy, wrapWithExitableDirectoryReader, executor, false);
}
```

Reply:
done.